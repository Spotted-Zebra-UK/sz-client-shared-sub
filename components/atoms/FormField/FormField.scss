@import '../../../variables';

.FormField {
  &__Label__Text {
    font-size: $p2-font-size;
    line-height: $small-line-height;
    text-align: left;
  }

  &__Label__Text--HasError {
    color: $purple;
  }

  &__Error {
    color: $purple;
    font-size: $p2-font-size;
    min-height: 16px;
    line-height: $small-line-height;
    margin-top: 8px;
    text-align: left;
  }
}
