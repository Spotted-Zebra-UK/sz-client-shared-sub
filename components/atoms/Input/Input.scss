@import '../../../variables';

.Input {
  border: 0;
  border-bottom: 2px solid $black;
  font-size: $p1-font-size;
  height: 36px;
  line-height: $medium-line-height;
  width: 100%;

  &:disabled {
    background-color: transparent;
  }
}
