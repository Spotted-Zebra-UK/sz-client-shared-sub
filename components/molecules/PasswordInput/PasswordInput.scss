@import '../../../variables';

.PasswordInput {
  border-bottom: 2px solid $black;
  display: flex;

  input {
    border: 0;
    font-size: $p1-font-size;
    height: 36px;
    line-height: $medium-line-height;
    width: calc(100% - 22px);
    font-size: 14px;
  }

  &__Button {
    border: 0;
    background-color: transparent;
    cursor: pointer;

    svg {
      width: 22px;

      path {
        fill: rgba(0, 0, 0, 0.16);
      }
    }
  }
}
