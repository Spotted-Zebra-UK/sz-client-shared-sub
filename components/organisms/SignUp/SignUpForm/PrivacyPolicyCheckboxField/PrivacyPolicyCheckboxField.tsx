import { FC } from 'react';
import { useTranslation } from 'react-i18next';
import { Link } from '@spotted-zebra-uk/sz-ui-shared.ui.link';
import { Checkbox } from '@spotted-zebra-uk/sz-ui-shared.widgets.checkbox';

interface IPrivacyPolicyCheckboxField {
  value: boolean;
  onChange: (value: boolean, name: string) => void;
  name: string;
  id: string;
  error?: string;
}

const PrivacyPolicyCheckboxField: FC<IPrivacyPolicyCheckboxField> = ({
  value,
  name,
  onChange,
  error,
  ...props
}) => {
  const { t } = useTranslation();
  const renderPrivacyPolicyCheckboxLabel = () => {
    return (
      <span className="SignUpForm__PrivacyPolicyCheckboxLabel">
        {t('authentication.signUp.iHaveReadThe')}
        <Link
          to="https://www.spottedzebra.co.uk/privacy/privacy"
          target="_blank"
        >
          {t('authentication.signUp.privacyNotice')}
        </Link>
        {t('authentication.signUp.iAmAnAdult')}
      </span>
    );
  };

  const handleChange = () => {
    onChange(!value, name);
  };

  return (
    <div className="CheckboxFormField">
      <Checkbox
        label={renderPrivacyPolicyCheckboxLabel()}
        checked={value}
        onChange={handleChange}
        hasError={!!error?.length}
        {...props}
        bottomText={error}
      />
    </div>
  );
};

export default PrivacyPolicyCheckboxField;
