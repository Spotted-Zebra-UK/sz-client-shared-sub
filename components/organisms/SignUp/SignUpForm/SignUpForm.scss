@import '../../../../variables';

.SignUpForm {
  .FormField--email,
  .FormField--fullName,
  .FormField--password {
    margin-bottom: 8px;
  }

  .FormField--confirmPassword {
    margin-bottom: 10px;
  }

  .SquareButton {
    display: block;
    margin: 0 auto;
  }

  .CheckboxFormField {
    margin-bottom: 24px;
  }

  &__PrivacyPolicyCheckboxLabel {
    font-size: $p2-font-size;

    &__Link {
      font-weight: 500;
      text-decoration: underline;
    }
  }
}
