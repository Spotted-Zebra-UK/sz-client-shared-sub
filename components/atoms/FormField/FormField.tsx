import './FormField.scss';
import React, { FunctionComponent } from 'react';

interface IFormField {
  error?: string | undefined;
  fieldName: string;
  label?: string;
  isLabelVisible?: boolean;
}

const FormField: FunctionComponent<IFormField> = ({
  children,
  error,
  label,
  fieldName,
  isLabelVisible = true,
}) => {
  return (
    <div className={`FormField FormField--${fieldName}`}>
      <label className="FormField__Label" htmlFor={fieldName}>
        <div
          className={`FormField__Label__Text${
            error ? ' FormField__Label__Text--HasError' : ''
          }`}
        >
          {label && isLabelVisible ? label : ''}
        </div>
        {children}
        <div className="FormField__Error">{error}</div>
      </label>
    </div>
  );
};

export default FormField;
