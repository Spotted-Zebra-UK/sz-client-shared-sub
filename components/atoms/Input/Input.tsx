import './Input.scss';
import React, { FunctionComponent } from 'react';

interface IInput {
  className?: string;
  name: string;
  onChange: (value: string, name: string) => void;
  placeholder?: string;
  value: string;
  type?: 'text' | 'password' | 'date' | 'number';
  isDisabled?: boolean;
  id?: string;
}

const Input: FunctionComponent<IInput> = ({
  className,
  name,
  onChange,
  placeholder,
  value,
  type,
  isDisabled,
  id,
}) => {
  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    onChange(event.target.value, name);
  };

  return (
    <input
      className={`Input${className ? ` ${className}` : ''}`}
      name={name}
      onChange={handleChange}
      onBlur={handleChange}
      value={value}
      type={type || 'text'}
      disabled={isDisabled || false}
      placeholder={placeholder}
      id={id}
    />
  );
};

export default Input;
