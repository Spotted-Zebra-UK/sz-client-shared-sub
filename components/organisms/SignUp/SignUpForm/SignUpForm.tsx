import './SignUpForm.scss';
import capitalize from 'lodash/capitalize';
import isEmpty from 'lodash/isEmpty';
import React, { FC, useState } from 'react';
import { useTranslation } from 'react-i18next';
import { Button } from '@spotted-zebra-uk/sz-ui-shared.ui.button';
import { TextInputField } from '@spotted-zebra-uk/sz-ui-shared.widgets.text-input-field';
import { passwordValidationRegex } from '../../../../constants/validation';
import { formatFullName } from '../../../../helpers/fullName';
import validate from '../../../../helpers/validate';
import PrivacyPolicyCheckboxField from './PrivacyPolicyCheckboxField/PrivacyPolicyCheckboxField';

interface ISignUpFormValues {
  fullName: string;
  email: string;
  password: string;
  isPrivacyPolicyChecked: boolean;
}

interface ISignUpFormErrors {
  fullName: string[];
  email: string[];
  password: string[];
  isPrivacyPolicyChecked: string[];
}

interface ISignUpForm {
  fullName?: string;
  email?: string;
  onSignUp: (
    fullName: string,
    email: string,
    password: string,
    isPrivacyPolicyChecked: boolean
  ) => void;
}

const SignUpForm: FC<ISignUpForm> = props => {
  const { t } = useTranslation();
  const [values, setValues] = useState<ISignUpFormValues>({
    fullName: props.fullName || '',
    email: props.email || '',
    password: '',
    isPrivacyPolicyChecked: false,
  });
  const [errors, setErrors] = useState<ISignUpFormErrors | undefined>();
  const loginValidate = () => {
    return validate(values, {
      fullName: {
        length: {
          minimum: 2,
          tooShort: `^${t(
            'authentication.signUp.firstNameAndLastNameMustBeProvided'
          )}`,
          tokenizer: (value: string) => {
            return formatFullName(value).split(/\s+/g);
          },
        },
      },
      email: {
        email: {
          message: `^${t('common.notValidEmail')}`,
        },
      },
      password: {
        format: {
          pattern: passwordValidationRegex,
          flags: 'i',
          message: `^${t('authentication.signUp.yourPasswordMustHave')}`,
        },
      },
      isPrivacyPolicyChecked: {
        exclusion: {
          within: { false: false },
          message: `^${t('authentication.signUp.privacyNoticeMustBeCached')}`,
        },
      },
    });
  };

  const handleChange = (value: string, name: string) => {
    setValues(prevValues => ({ ...prevValues, [name]: value }));
    setErrors(prevErrors =>
      prevErrors ? { ...prevErrors, [name]: [] } : prevErrors
    );
  };

  const handleChangePrivacyPolicyCheck = () => {
    setValues(prevValues => ({
      ...prevValues,
      isPrivacyPolicyChecked: !prevValues.isPrivacyPolicyChecked,
    }));
    setErrors(prevErrors =>
      prevErrors ? { ...prevErrors, isPrivacyPolicyChecked: [] } : prevErrors
    );
  };

  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault();
    const newErrors = loginValidate();
    if (newErrors) {
      setErrors(newErrors);
    } else {
      props.onSignUp(
        formatFullName(values.fullName),
        values.email,
        values.password,
        values.isPrivacyPolicyChecked
      );
    }
  };

  return (
    <form className="SignUpForm" onSubmit={handleSubmit}>
      <div className="SignUpForm__Fields">
        <TextInputField
          id="fullName"
          label={t('authentication.signUp.fullName')}
          value={values.fullName}
          placeholder={t('authentication.signUp.fullName')}
          onChange={handleChange}
          ariaLabel={t('authentication.signUp.fullName')}
          hasError={!isEmpty(errors?.fullName)}
          bottomText={errors?.fullName?.join(' ')}
          disabled={!!props.fullName?.length}
        />
        <TextInputField
          id="email"
          label={capitalize(t('common.email'))}
          value={values.email}
          placeholder={capitalize(t('common.email'))}
          onChange={handleChange}
          ariaLabel={t('common.email')}
          hasError={!isEmpty(errors?.email)}
          bottomText={errors?.email?.join(' ')}
          disabled={!!props.email?.length}
          type="email"
        />
        <TextInputField
          id="password"
          label={t('authentication.signUp.createPassword')}
          value={values.password}
          placeholder={capitalize(t('common.password'))}
          onChange={handleChange}
          ariaLabel={t('authentication.signUp.createPassword')}
          hasError={!isEmpty(errors?.password)}
          bottomText={errors?.password?.join(' ')}
          type="password"
        />
      </div>
      <div className="SignUpForm__Bottom">
        <PrivacyPolicyCheckboxField
          value={values.isPrivacyPolicyChecked}
          onChange={handleChangePrivacyPolicyCheck}
          name="isPrivacyPolicyChecked"
          id="SignUpIsPrivacyPolicyChecked"
          error={errors?.isPrivacyPolicyChecked?.join(' ')}
        />
        <Button type="submit" className="SignUpForm__SubmitButton">
          {t('authentication.signUp.createAccount')}
        </Button>
      </div>
    </form>
  );
};

export default SignUpForm;
