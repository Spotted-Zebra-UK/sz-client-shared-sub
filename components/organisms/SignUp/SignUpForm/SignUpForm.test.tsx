import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import FormField from '../../../atoms/FormField/FormField';
import Input from '../../../atoms/Input/Input';
import PasswordInput from '../../../molecules/PasswordInput/PasswordInput';
import PrivacyPolicyCheckboxField from './PrivacyPolicyCheckboxField/PrivacyPolicyCheckboxField';
import SignUpForm from './SignUpForm';

describe('SignUpForm', () => {
  let onSignUp: (
    fullName: string,
    email: string,
    password: string,
    isPrivacyPolicyChecked: boolean
  ) => void;

  const fullNameValue = 'test-firstName lastName';
  const passwordValue = 'test-passwordValue';
  const emailValue = 'test@email.com';

  beforeEach(() => {
    onSignUp = jest.fn();
  });

  it('should render correctly by default', () => {
    const wrapper = shallow(<SignUpForm onSignUp={onSignUp} />);

    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find(Input).at(0).props().value).toBe('');
    expect(wrapper.find(FormField).at(0).props().error).toBeUndefined();
    expect(wrapper.find(Input).at(1).props().value).toBe('');
    expect(wrapper.find(FormField).at(1).props().error).toBeUndefined();
    expect(wrapper.find(PasswordInput).props().value).toBe('');
    expect(wrapper.find(FormField).at(2).props().error).toBeUndefined();
  });

  it('should change password input value on input change', () => {
    const wrapper = shallow(<SignUpForm onSignUp={onSignUp} />);

    wrapper.find(PasswordInput).props().onChange(passwordValue, 'password');
    expect(wrapper.find(PasswordInput).props().value).toBe(passwordValue);
  });

  it('should change email input value on input change', () => {
    const wrapper = shallow(<SignUpForm onSignUp={onSignUp} />);

    wrapper.find(Input).at(1).props().onChange(emailValue, 'email');
    expect(wrapper.find(Input).at(1).props().value).toBe(emailValue);
  });

  it('should change fullName input value on input change', () => {
    const wrapper = shallow(<SignUpForm onSignUp={onSignUp} />);

    wrapper.find(Input).at(0).props().onChange(fullNameValue, 'fullName');
    expect(wrapper.find(Input).at(0).props().value).toBe(fullNameValue);
  });

  it('should change privacy policy checkbox value on checkbox change', () => {
    const wrapper = shallow(<SignUpForm onSignUp={onSignUp} />);

    wrapper
      .find(PrivacyPolicyCheckboxField)
      .props()
      .onChange(true, 'isPrivacyPolicyChecked');
    expect(wrapper.find(PrivacyPolicyCheckboxField).props().value).toBe(true);
  });

  it('should call provided onSignUp', () => {
    const wrapper = shallow(<SignUpForm onSignUp={onSignUp} />);

    wrapper.find(PasswordInput).props().onChange(passwordValue, 'password');
    wrapper.find(Input).at(0).props().onChange(fullNameValue, 'fullName');
    wrapper.find(Input).at(1).props().onChange(emailValue, 'email');
    wrapper
      .find(PrivacyPolicyCheckboxField)
      .props()
      .onChange(true, 'isPrivacyPolicyChecked');

    const eventMock = {
      preventDefault: jest.fn(),
    };

    wrapper.find('form').simulate('submit', eventMock);

    expect(eventMock.preventDefault).toHaveBeenCalled();
    expect(onSignUp).toHaveBeenCalledWith(
      fullNameValue,
      emailValue,
      passwordValue,
      true
    );
  });

  it('should show error and not call onSignUp if submit single string fullName', () => {
    const wrapper = shallow(<SignUpForm onSignUp={onSignUp} />);

    const eventMock = {
      preventDefault: jest.fn(),
    };

    wrapper.find(Input).at(0).props().onChange('firstName', 'fullName');

    wrapper.find('form').simulate('submit', eventMock);

    expect(wrapper.find(FormField).at(0).props().error).toBe(
      'First name and last name must be provided.'
    );
  });

  it('should show error and not call onSignUp if submit too short password', () => {
    const wrapper = shallow(<SignUpForm onSignUp={onSignUp} />);

    wrapper.find(Input).at(0).props().onChange(fullNameValue, 'fullName');
    wrapper.find(Input).at(1).props().onChange(emailValue, 'email');
    wrapper
      .find(PrivacyPolicyCheckboxField)
      .props()
      .onChange(true, 'isPrivacyPolicyChecked');

    const eventMock = {
      preventDefault: jest.fn(),
    };

    wrapper.find('form').simulate('submit', eventMock);

    expect(eventMock.preventDefault).toHaveBeenCalled();
    expect(wrapper.find(FormField).at(2).props().error).toBe(
      'Password is too short (minimum is 5 characters)'
    );
    expect(onSignUp).not.toHaveBeenCalled();
  });

  it('should show error and not call onSignUp if submit invalid email', () => {
    const wrapper = shallow(<SignUpForm onSignUp={onSignUp} />);

    wrapper.find(PasswordInput).props().onChange(passwordValue, 'password');
    wrapper.find(Input).at(0).props().onChange(fullNameValue, 'fullName');
    wrapper.find(Input).at(1).props().onChange('invalid-emial-value', 'email');
    wrapper
      .find(PrivacyPolicyCheckboxField)
      .props()
      .onChange(true, 'isPrivacyPolicyChecked');

    const eventMock = {
      preventDefault: jest.fn(),
    };

    wrapper.find('form').simulate('submit', eventMock);

    expect(eventMock.preventDefault).toHaveBeenCalled();
    expect(wrapper.find(FormField).at(1).props().error).toBe(
      'Not a valid email.'
    );
    expect(onSignUp).not.toHaveBeenCalled();
  });

  it('should show error and not call onSignUp if submit isPrivacyPolicyChecked is not checked', () => {
    const wrapper = shallow(<SignUpForm onSignUp={onSignUp} />);

    wrapper.find(PasswordInput).props().onChange(passwordValue, 'password');
    wrapper.find(Input).at(0).props().onChange(fullNameValue, 'fullName');
    wrapper.find(Input).at(1).props().onChange(emailValue, 'email');

    const eventMock = {
      preventDefault: jest.fn(),
    };

    wrapper.find('form').simulate('submit', eventMock);

    expect(eventMock.preventDefault).toHaveBeenCalled();
    expect(wrapper.find(PrivacyPolicyCheckboxField).props().error).toBe(
      'Privacy Notice must be cheked.'
    );
    expect(onSignUp).not.toHaveBeenCalled();
  });

  it('should hide password error if error is visible and password input changed', () => {
    const wrapper = shallow(<SignUpForm onSignUp={onSignUp} />);

    const eventMock = {
      preventDefault: jest.fn(),
    };

    wrapper.find('form').simulate('submit', eventMock);

    expect(wrapper.find(FormField).at(2).props().error).toBe(
      'Password is too short (minimum is 5 characters)'
    );

    wrapper.find(PasswordInput).props().onChange(passwordValue, 'password');
    expect(wrapper.find(FormField).at(2).props().error).toBe('');
  });

  it('should hide email error if error is visible and email input changed', () => {
    const invalidEmailValue = 'test-emailValue';
    const wrapper = shallow(<SignUpForm onSignUp={onSignUp} />);

    wrapper.find(Input).at(1).props().onChange(invalidEmailValue, 'email');

    const eventMock = {
      preventDefault: jest.fn(),
    };

    wrapper.find('form').simulate('submit', eventMock);

    expect(wrapper.find(FormField).at(1).props().error).toBe(
      'Not a valid email.'
    );

    wrapper.find(Input).at(1).props().onChange(emailValue, 'email');
    expect(wrapper.find(FormField).at(1).props().error).toBe('');
  });

  it('should hide fullName error if error is visible and fullName input changed', () => {
    const wrapper = shallow(<SignUpForm onSignUp={onSignUp} />);

    const eventMock = {
      preventDefault: jest.fn(),
    };

    wrapper.find('form').simulate('submit', eventMock);

    expect(wrapper.find(FormField).at(0).props().error).toBe(
      'First name and last name must be provided.'
    );

    wrapper.find(Input).at(1).props().onChange(fullNameValue, 'fullName');
    expect(wrapper.find(FormField).at(0).props().error).toBe('');
  });

  it('should show error and not call onSignUp if submit invalid email', () => {
    const wrapper = shallow(<SignUpForm onSignUp={onSignUp} />);

    const eventMock = {
      preventDefault: jest.fn(),
    };

    wrapper.find('form').simulate('submit', eventMock);

    expect(eventMock.preventDefault).toHaveBeenCalled();
    expect(wrapper.find(PrivacyPolicyCheckboxField).props().error).toBe(
      'Privacy Notice must be cheked.'
    );
    expect(onSignUp).not.toHaveBeenCalled();

    wrapper
      .find(PrivacyPolicyCheckboxField)
      .props()
      .onChange(true, 'isPrivacyPolicyChecked');
    expect(wrapper.find(PrivacyPolicyCheckboxField).props().error).toBe('');
  });

  it('should not render fullName input when fullName provided', () => {
    const wrapper = shallow(
      <SignUpForm onSignUp={onSignUp} fullName={fullNameValue} />
    );

    expect(wrapper.find('[fieldName="fullName"]')).toHaveLength(0);
    expect(wrapper.find('[name="fullName"]')).toHaveLength(0);
    expect(wrapper.find(Input).at(0).props().name).toBe('email');
  });

  it('should call onSubmit on submit when fullName provided', () => {
    const wrapper = shallow(
      <SignUpForm onSignUp={onSignUp} fullName={fullNameValue} />
    );

    wrapper.find(PasswordInput).props().onChange(passwordValue, 'password');
    wrapper.find(Input).at(0).props().onChange(emailValue, 'email');
    wrapper
      .find(PrivacyPolicyCheckboxField)
      .props()
      .onChange(true, 'isPrivacyPolicyChecked');

    const eventMock = {
      preventDefault: jest.fn(),
    };

    wrapper.find('form').simulate('submit', eventMock);

    expect(eventMock.preventDefault).toHaveBeenCalled();
    expect(onSignUp).toHaveBeenCalledWith(
      fullNameValue,
      emailValue,
      passwordValue,
      true
    );
  });

  it('should disable email input when email provided', () => {
    const wrapper = shallow(
      <SignUpForm
        onSignUp={onSignUp}
        fullName={fullNameValue}
        email={emailValue}
      />
    );

    expect(wrapper.find('[fieldName="fullName"]')).toHaveLength(0);
    expect(wrapper.find('[name="fullName"]')).toHaveLength(0);
    expect(wrapper.find(Input).at(0).props().isDisabled).toBe(true);
    expect(wrapper.find(Input).at(0).props().name).toBe('email');
  });

  it('should call onSubmit on submit when fullName and email provided', () => {
    const wrapper = shallow(
      <SignUpForm
        onSignUp={onSignUp}
        fullName={fullNameValue}
        email={emailValue}
      />
    );

    wrapper.find(PasswordInput).props().onChange(passwordValue, 'password');
    wrapper
      .find(PrivacyPolicyCheckboxField)
      .props()
      .onChange(true, 'isPrivacyPolicyChecked');

    const eventMock = {
      preventDefault: jest.fn(),
    };

    wrapper.find('form').simulate('submit', eventMock);

    expect(eventMock.preventDefault).toHaveBeenCalled();
    expect(onSignUp).toHaveBeenCalledWith(
      fullNameValue,
      emailValue,
      passwordValue,
      true
    );
  });
});
