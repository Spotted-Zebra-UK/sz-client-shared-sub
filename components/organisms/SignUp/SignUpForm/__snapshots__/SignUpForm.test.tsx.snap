// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpForm should render correctly by default 1`] = `
<form
  className="SignUpForm"
  onSubmit={[Function]}
>
  <div
    className="SignUpForm__Fields"
  >
    <FormField
      fieldName="fullName"
      isLabelVisible={false}
      label="Full name"
    >
      <Input
        name="fullName"
        onChange={[Function]}
        placeholder="Full name"
        value=""
      />
    </FormField>
    <FormField
      fieldName="email"
      isLabelVisible={false}
      label="Email"
    >
      <Input
        isDisabled={false}
        name="email"
        onChange={[Function]}
        placeholder="Email"
        value=""
      />
    </FormField>
    <FormField
      fieldName="password"
      isLabelVisible={false}
      label="Create Password"
    >
      <PasswordInput
        name="password"
        onChange={[Function]}
        placeholder="Password"
        value=""
      />
    </FormField>
  </div>
  <div
    className="SignUpForm__Bottom"
  >
    <PrivacyPolicyCheckboxField
      id="SignUpIsPrivacyPolicyChecked"
      name="isPrivacyPolicyChecked"
      onChange={[Function]}
      value={false}
    />
    <SquareButton
      type="submit"
    >
      Create an account
    </SquareButton>
  </div>
</form>
`;
